source/common/network/apple_dns_impl.cc

```diff
@@ -93,6 +113,9 @@ void AppleDnsResolverImpl::initializeMainSdRef() {
       [this](uint32_t events) { onEventCallback(events); }, Event::FileTriggerType::Level,
       Event::FileReadyType::Read);
   sd_ref_event_->setEnabled(Event::FileReadyType::Read);
+
+  initialize_failure_timer_->disableTimer();
+  backoff_strategy_->reset();
```

Comment:
Is this the case where we have a back off ongoing, someone else comes in here, and they succeed, so we can cancel the ongoing backoff? More comments?

Reply:
yep exactly! Yeah will add more comments.